config/constants.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Constants
+  SEED_ADMIN_EMAIL = ENV.fetch("SEED_ADMIN_EMAIL", nil)
+  SEED_ADMIN_INITIAL_PASSWORD = ENV.fetch("SEED_ADMIN_INITIAL_PASSWORD", nil)
+  MAIL_FROM = ENV.fetch("MAIL_FROM", nil)
+  SIDEKIQ_USERNAME = ENV.fetch("SIDEKIQ_USERNAME", nil)
+  SIDEKIQ_PASSWORD = ENV.fetch("SIDEKIQ_PASSWORD", nil)
+  HTTP_AUTH_USERNAME = ENV.fetch("HTTP_AUTH_USERNAME", nil)
+  HTTP_AUTH_PASSWORD = ENV.fetch("HTTP_AUTH_PASSWORD", nil)
+  LOCALE = ENV.fetch("LOCALE", "en")
+  ROLLBAR_ACCESS_TOKEN = ENV.fetch("ROLLBAR_ACCESS_TOKEN", nil)
+  ROLLBAR_ENV = ENV.fetch("ROLLBAR_ENV", nil)
+  REDIS_URL = ENV.fetch("REDIS_URL", nil)
+  DISABLE_WEBDRIVERS = ENV.fetch("DISABLE_WEBDRIVERS", nil)
+  CAPYBARA_DRIVER = ENV.fetch("CAPYBARA_DRIVER", nil)
+  TEST_ENV_NUMBER = ENV.fetch("TEST_ENV_NUMBER", nil)
+  ENABLE_RACK_ATTACK = ENV.fetch("ENABLE_RACK_ATTACK", nil)
+  BLOCKED_IPS = ENV.fetch("BLOCKED_IPS", nil)
+  BACKTRACE = ENV.fetch("BACKTRACE", nil)
+  RAILS_SERVE_STATIC_FILES = ENV.fetch("RAILS_SERVE_STATIC_FILES", nil)
+  USE_LETTER_OPENER = ENV.fetch("USE_LETTER_OPENER", nil)
+  ALLOWED_HOSTS = ENV.fetch("ALLOWED_HOSTS", "")
+  RAILS_LOG_TO_STDOUT = ENV.fetch("RAILS_LOG_TO_STDOUT", nil)
+  SENDINBLUE_EMAIL = ENV.fetch("SENDINBLUE_EMAIL", nil)
+  SENDINBLUE_PASSWORD = ENV.fetch("SENDINBLUE_PASSWORD", nil)
+  DOMAIN_NAME = ENV.fetch("DOMAIN_NAME", nil)
+  HEROKU_APP_NAME = "#{ENV.fetch('HEROKU_APP_NAME', nil)}.herokuapp.com".freeze
```

Comment:
```suggestion
  HEROKU_APP_NAME = "#{ENV.fetch('HEROKU_APP_NAME')}.herokuapp.com".freeze
```

Reply:
HEROKU_APP_NAME is something heroku provides for review apps.
In the production config we have some logic around for setting the host name to HEROKU_APP_NAME only if it's available. Otherwise we use the DOMAIN_NAME var.

It is a mistake that the string templating was moved to `constants.rb` however, since it means that the app will always have a HEROKU_APP_NAME, which is not the purpose.

The question is whether or not we want to keep this functionality around when we are not currently using review apps AFAIK on any recent projects.

Reply:
I believe `HEROKU_APP_NAME` is given to all apps https://devcenter.heroku.com/articles/dyno-metadata
I vaguely remember seeing an addon using it (can't remember whether it would work if the name was nullified).

I agree that it is better to reduce name usage to `DOMAIN_NAME` unless we have a good reason. This will reduce possible differences between the value of `HEROKU_APP_NAME` in Heroku environment and the app environment.

Reply:
Not sure what is the conclusion here, so I've kept it as it is. Do we want to remove it from `constants.rb` AND `production.rb` ?